src/task/TraitManager.ts

```diff
@@ -35,16 +37,29 @@ export class TraitManager {
 		}
 		return Promise.resolve(completions);
 	}
+	static async provideTraitProperties(traitName: string, position: vscode.Position): Promise<vscode.CompletionItem[]> {
+		const completions: vscode.CompletionItem[] = [];
+		const traitDefs = await TraitManager.retrieveTraitsDefinitions(traitName);
+		for (const property of traitDefs[0].properties) {
```

Comment:
There is also a `foreach` available. It might be easier to use and read when using it to loop the properties. I'd also recommend to specify types if possible.

Something like...
`traitDefs[0].properties.forEach(` (element: <WhatEverTypePropertiesContains>) => {
   // do something with element
});

Reply:
seems quite equivalent to read in this case.
I think it is useful if treatments are "streamed"

Changed to use forEach, it will be more consistent if we are using "streamed" modifications.

Reply:
My goal was just to show you this alternative because I was rather unsure you knew about it.

Reply:
ok, thanks